cores/esp32/esp32-hal-i2c.c

```diff
@@ -306,9 +307,12 @@ static void i2cDumpDqData(i2c_t * i2c)
         }
         a++;
     }
+#else
+    log_i("Debug Buffer not Enabled");
```

Comment:
it would be better to not log anything at all here so as not to clutter the console

Reply:
Except, if debug level is at "info" I assume you actually want to see information? 
The debug buffer in hal-i2c takes up 2570 bytes of ram if enabled. 
That is why this message exists. If you set debug_level at or above "info" I expect you have a problem you are trying to solve.

Reply:
@stickbreaker this function isn't called from what I can tell if the level isn't at least DEBUG.

Reply:
I have tried to follow same as others functions - should I change it ?

Reply:
My bad, I got confused with the add/subtract highlighting. I just though you were questioning the "enable" message.  My only excuse is that I hadn't had my coffee before I read your Post :smiley: 

I don't have any problems with your changes.

Reply:
coffee is a must!

Reply:
One other question on the debug subject: I added a an app level control for debug output with `Wire.setDebugFlags( uint32_t setBits, uint32_t resetBits);`, but I did not document how/when to use it.  Is there any standard for such documentation?

I used it around calls to a "recalcitrant" i2c devices when I get unexpected responses. I don't know if it would be good to put "debug" conditionals around all levels of this code (i.e. Wire() and hal-i2c).

What do you think?

Reply:
@stickbreaker @atanisoft  I followed same as line 354 - it means, I need to remove original 354 as well ? Or I misunderstood ?
thanks for your review

Reply:
@luc-github Let me spend some time looking through it.  The "debug_buffer" only houses interrupt activity(used inside the ISR). I just scanned through `i2cDumpDqData()` and don't know why I had it conditional on "debug_buffer"?

Reply:
For some reason github won't let me create an edited version of your version? So, I'm going to create an edited version the current espressif version in my repo? (might work? very confusing to me anyway!)